### PR TITLE
Code quality improvements to WordPressComServiceRemote

### PR DESCRIPTION
**Pull request by diegoreymendez**
This PR improves `WordPressComServiceRemote` by:

- Documenting its methods.
- Naming the block types used.
- Adding a few missing assertions.
- The assertions added made me realize that even though we were treating `languageId` as a number, it was actually a string object.  I fixed the data type.

**To test:**
1. Launch the app
2. At the login screen tap on "Create account"
3. Fill the forms
4. Make sure the account and blog are created.

/cc @jleandroperez

### CODE REVIEW

**Comment by diegoreymendez** [comment]
Changed the name of the PR since it was a bit outdated.

**Comment by diegoreymendez** [comment]
Travis mentions 1 test is failing but I'm not being able to make any test fail locally.

File: WordPress/Classes/Networking/WordPressComServiceRemote.h [hunk 1]
```diff
@@ -6,25 +6,60 @@ typedef NS_ENUM(NSUInteger, WordPressComServiceBlogVisibility) {
...
+ *  @param      password    The password of the new account.  Cannot be nil.
+ *  @param      success     The block to execute on success.  Can be nil.
+ *  @param      failure     The block to execute on failure.  Can be nil.
+ */
 - (void)createWPComAccountWithEmail:(NSString *)email
                         andUsername:(NSString *)username
                         andPassword:(NSString *)password
-                            success:(void (^)(id responseObject))success
-                            failure:(void (^)(NSError *error))failure;
+                            success:(WordPressComServiceSuccessBlock)success
```
Comment (jleandroperez) [inline WordPress/Classes/Networking/WordPressComServiceRemote.h]: MUCH better, thanks!!!

File: WordPress/Classes/Networking/WordPressComServiceRemote.m [hunk 2]
```diff
@@ -12,22 +12,31 @@ @implementation WordPressComServiceRemote
...
                         andPassword:(NSString *)password
-                            success:(void (^)(id responseObject))success
-                            failure:(void (^)(NSError *error))failure
+                            success:(WordPressComServiceSuccessBlock)success
+                            failure:(WordPressComServiceFailureBlock)failure
 {
-    [self createWPComAccountWithEmail:email andUsername:username andPassword:password validate:NO success:success failure:failure];
+    NSParameterAssert([email isKindOfClass:[NSString class]]);
+    NSParameterAssert([username isKindOfClass:[NSString class]]);
+    NSParameterAssert([password isKindOfClass:[NSString class]]);
```
Comment (jleandroperez) [inline WordPress/Classes/Networking/WordPressComServiceRemote.m]: Asserts FTW!

**Comment by jleandroperez** [comment]
@diegoreymendez no worries! We should remove Travis (IMHO), since it's always giving off false alarms.

Heads up, i've found few warnings in `CreateAccountAndBlogViewController`:

screen shot 2015-08-24 at 12 26 49 pm

**Comment by diegoreymendez** [comment]
@jleandroperez - Good catch.  Xcode wasn't showing those warnings here until I went for a clean & build.

It's ready for another look!

**Comment by jleandroperez** [comment]
Looking great @diegoreymendez!

:shipit:

**Comment by diegoreymendez** [comment]
Ty @jleandroperez !

**Pull request merged**